windows/CodePush.cs

```diff
@@ -0,0 +1,668 @@
+        }
+
+        public async Task ClearUpdates()
+        {
+            await codePushPackage.ClearUpdates();
+            RemovePendingUpdate();
+            RemoveFailedUpdates();
+        }
+
+        private class CodePushNativeModule : ReactContextNativeModuleBase
+        {
+            private ILifecycleEventListener lifecycleEventListener = null;
+            private int minimumBackgroundDuration = 0;
+            private ReactContext reactContext;
+
+            private class CodePushResumeListener : ILifecycleEventListener
+            {
+                private DateTime? lastPausedDate = null;
+                private CodePushNativeModule codePushNativeModule;
+
+                public CodePushResumeListener(CodePushNativeModule codePushNativeModule)
+                {
+                    this.codePushNativeModule = codePushNativeModule;
+                }
+
+                public void OnDestroy()
+                {
+                }
+
+                public void OnResume()
+                {
+                    if (lastPausedDate != null)
+                    {
+                        // Determine how long the app was in the background and ensure
+                        // that it meets the minimum duration amount of time.
+                        double durationInBackground = (new DateTime() - (DateTime)lastPausedDate).TotalSeconds;
+                        if (durationInBackground >= codePushNativeModule.minimumBackgroundDuration)
+                        {
+                            Action loadBundleAction = async () =>
+                            {
+                                await codePushNativeModule.LoadBundle();
+                            };
+
+                            codePushNativeModule.Context.RunOnNativeModulesQueueThread(loadBundleAction);
+                        }
+                    }
+                }
+
+                public void OnSuspend()
+                {
+                    // Save the current time so that when the app is later
+                    // resumed, we can detect how long it was in the background.
+                    lastPausedDate = new DateTime();
+                }
+            }
+
+            private CodePushModule codePush;
+
+            public CodePushNativeModule(ReactContext reactContext, CodePushModule codePush) : base(reactContext)
+            {
+                this.reactContext = reactContext;
+                this.codePush = codePush;
+            }
+
+            public override string Name
+            {
+                get
+                {
+                    return "CodePush";
+                }
+            }
+
+            public override IReadOnlyDictionary<string, object> Constants
+            {
+                get
+                {
+                    Dictionary<string, object> constants = new Dictionary<string, object>();
+                    constants["codePushInstallModeImmediate"] = CodePushInstallMode.IMMEDIATE;
```

Comment:
Should we use a [dictionary initializer](https://msdn.microsoft.com/en-us/library/bb531208.aspx) for this?